src/main/haskell/kore/src/Kore/Unification/SubstitutionNormalization.hs

```diff
@@ -82,8 +82,12 @@ normalizeSubstitution tools substitution = do
     interestingVariables = extractVariables substitution
     variableToPattern = Map.fromList substitution
     dependencies = buildDependencies substitution interestingVariables
+    depsWithoutReflexivity = Map.filterWithKey (notReflexive) dependencies
+    notReflexive :: (Eq a, Show a) => a -> [a] -> Bool
+    notReflexive a [x]= a /= x
```

Comment:
This is a quick reply from my mobile phone, so I may be wrong.

This does not seem right.

The following must happen when calling normalizeSubstitution (in Kore.Substitution.SubstitutionNormalization):

`[a=a]` --> `[]`

`[a=f(a)]` --> `error`

`[a=a,a=b]` --> `[a=b]`

`[a=b, b=f(a)]`>`error`

I don't think that the second and the third are working now. BTW, you should probably add these as tests for normalizeSubstitution.